omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -82,6 +83,13 @@ public void setSessionLocation(Location sessionLocation) {
         }
         this.sessionLocation = sessionLocation;
         this.sessionLocationId = sessionLocation.getId();
+        setSessionLocationInUserContext(sessionLocation);
```

Comment:
Do you have to set it again here after the above?

Reply:
I thought, it would be clean to have it while making initialization

Reply:
My point is, do you need to do it twice?

Reply:
Yah, No need it twice, I have removed it from here.